### PR TITLE
[#173382894] Ruby 2.7

### PR DESCRIPTION
**Pull request by tlwr**
[Story](https://www.pivotaltracker.com/story/show/173382894)

What
----

- Use ruby 2.7.1
- Use Ubuntu Focal for running the tests
- Update Rubocop
- Rename some of the rubocops since they changed
- Do not use govuk-lint, as it is deprecated, there is a rubocop replacement
- Do not run `lint_ruby` as part of `lint` and therefore `test` make task

In a follow up PR I will fix all the linting errors, so this one is easier to review

How to review
-------------

CI

Check Travis is still running the tests, and linting

Maybe run on your development environment pipeline

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -11,20 +11,24 @@ DEPLOY_ENV_VALID_CHARS=$(shell if echo $(DEPLOY_ENV) | grep -q '^[a-zA-Z0-9-]*$$
 LOGSEARCH_BOSHRELEASE_TAG=v209.0.0
 LOGSEARCH_FOR_CLOUDFOUNDRY_TAG=v207.0.0
 
+.PHONY:	check-env
```
Comment (AP-Hunt) [inline Makefile:14]: Extra space

File: Makefile [hunk 2]
```diff
@@ -11,20 +11,24 @@ DEPLOY_ENV_VALID_CHARS=$(shell if echo $(DEPLOY_ENV) | grep -q '^[a-zA-Z0-9-]*$$
...
 check-env:
 	$(if ${DEPLOY_ENV},,$(error Must pass DEPLOY_ENV=<name>))
 	$(if ${DEPLOY_ENV_VALID_LENGTH},,$(error Sorry, DEPLOY_ENV ($(DEPLOY_ENV)) has a max length of $(DEPLOY_ENV_MAX_LENGTH), otherwise derived names will be too long))
 	$(if ${DEPLOY_ENV_VALID_CHARS},,$(error Sorry, DEPLOY_ENV ($(DEPLOY_ENV)) must use only alphanumeric chars and hyphens, otherwise derived names will be malformatted))
 	$(if ${MAKEFILE_ENV_TARGET},,$(error Must set MAKEFILE_ENV_TARGET))
 	@./scripts/validate_aws_credentials.sh
 
-test: spec compile_platform_tests lint_yaml lint_terraform lint_shellcheck lint_concourse lint_ruby lint_posix_newlines lint_symlinks ## Run linting tests
+.PHONY: test
```
Comment (AP-Hunt) [inline Makefile:22]: Extra newline

Comment (tlwr) [reply Makefile:22]: This is github UI not an actual newline

File: Makefile (hunk 1 above)
Comment (tlwr) [reply Makefile]: This was a tab, fixed in most recent commit - good spot

**Pull request merged**